### DISCUSSION TITLE
Errors

### DISCUSSION THREAD

**Discussion by accumail** [petrovvlad/freeLib#119] [General]
ошибка устранена

**Comment by accumail** [comment]
Ну и что дальше....?

**Reply by petrovvlad** [reply]
Дальше… искать причину ошибок. Для начала надо узнать - какая ОС и её версия?

**Reply by accumail** [reply]
Ubuntu 20.04 со всеми обновлениями (голая только что установлена). Стоит она виртуалке WMWare WSPro 17.2 последняя, которая в свою очередь на сервере WS 2022LTSC. Хочу поставить на нее OPDS server. Сейчас использую freeLib 5.03 под windows, который время от времени падает и приходится его вручную перегружать.
Просто надоело это. И еще: если не сложно, сделайте нативную поддержку https. Чтобы не городить "ни к селу , ни к городу" обратный proxy server на nginx.

**Comment by petrovvlad** [comment]
Поправил. Были проблемы совместимости с Qt<5.14. Проверил на виртуалке (Ubuntu 20.04), у меня собралось. Если будут проблемы, пишите.

**Comment by accumail** [comment]
Привет!
Да, сейчас все нормально.
Перешел на Ваш форк freeLib.
Работает и индексирует базу мин в два раза быстрее! Ему бы еще нативную поддержку  https! А иначе это полупродукт, по моим понятиям. Даже в убогой Calibre от индуса такое есть.  Спасибо и дальнейших успехов вам в работе.

**Reply by petrovvlad** [reply]
https можно попробовать прикрутить.

**Reply by accumail** [reply]
Тогда и донат проплачу 300р, за такое не жалко. Спасибо.

**Reply by petrovvlad** [reply]
Я против доната. Проектом занимаюсь в свободное время, поэтому по срокам ничего не обещаю. 
И не спешите радоваться, https будет только для Qt ≥6.4.

**Reply by accumail** [reply]
Нет, спасибо, но Ubuntu 22.04 ставить не буду. А в Ubuntu 20.04 только QT5. А какая Вам разница: на QT5 или QT6 это внедрять? Ведь подавляющее большинство вообще не считает нужным переходить на Ubuntu 22.04 и относится к ней скорее негативно.

**Reply by petrovvlad** [reply]
Сейчас в freeLib две версии opds сервера:
1. На основе QTcpServer. Глючная, я не смог её исправить. Собирается для Qt<6.4. В будущем будет удалена.
2. На основе QHttpServer.  Работает стабильно. Собирается для Qt≥6.4 т.к. QHttpServer повился в 6.4.

Т.е. установка Ubuntu 22.04 не поможет, она идёт с Qt 6.2. Я стараюсь поддерживать старые библиотеки, но в данном случае не буду.

**Reply by petrovvlad** [reply]
С чем связан негатив к Ubuntu 22.04?
